tests/drum/unit/model_metadata/test_model_metadata.py

```diff
@@ -853,3 +862,141 @@ def test_num_col_values(condition, value, fails):
             NumColumns(condition, value)
     else:
         NumColumns(condition, value)
+
+
+def test_read_model_metadata_properly_casts_typeschema(tmp_path, training_metadata_yaml):
+    config_yaml = training_metadata_yaml + dedent(
+        """
+        typeSchema:
+           input_requirements:
+           - field: number_of_columns
+             condition: IN
+             value:
+               - 1
+               - 2
+           - field: data_types
+             condition: EQUALS
+             value:
+               - NUM
+               - TXT
+           output_requirements:
+           - field: number_of_columns
+             condition: IN
+             value: 2
+           - field: data_types
+             condition: EQUALS
+             value: NUM
+        """
+    )
+    with open(os.path.join(tmp_path, MODEL_CONFIG_FILENAME), mode="w") as f:
+        f.write(config_yaml)
+
+    yaml_conf = read_model_metadata_yaml(tmp_path)
+    output_reqs = yaml_conf["typeSchema"]["output_requirements"]
+    input_reqs = yaml_conf["typeSchema"]["input_requirements"]
+
+    value_key = "value"
+    expected_as_int_list = next(
+        (el for el in input_reqs if el["field"] == "number_of_columns")
+    ).get(value_key)
+    expected_as_str_list = next((el for el in input_reqs if el["field"] == "data_types")).get(
+        value_key
+    )
+    expected_as_int = next((el for el in output_reqs if el["field"] == "number_of_columns")).get(
+        value_key
+    )
+    expected_as_str = next((el for el in output_reqs if el["field"] == "data_types")).get(value_key)
+
+    assert all(isinstance(el, int) for el in expected_as_int_list)
+    assert all(isinstance(el, str) for el in expected_as_str_list)
+    assert isinstance(expected_as_str_list, list)
+
+    assert isinstance(expected_as_int, int)
+    assert isinstance(expected_as_str, str)
+
+
+@pytest.mark.parametrize(
+    "target_type, predictor_cls",
+    itertools.product([TargetType.TRANSFORM,], [PythonPredictor, RPredictor, JavaPredictor],),
+)
```

Comment:
Interesting way to create a parameterized list! I hadn't seen that before. I wonder if Zach would approve :)

Reply:
I personally prefer two separate `pytest.mark.parametrize`s